[RN 0.71.0-rc.5] newArchEnabled Project with path ':ReactAndroid' could not be found in project

`react-native-safe-area-context` builds fine w `newArchEnabled=false` but fails w `newArchEnabled=true`

```console
FAILURE: Build failed with an exception.

* Where:
Build file '/home/runner/work/RN071RC5/RN071RC5/node_modules/react-native-safe-area-context/android/build.gradle' line: 103

* What went wrong:
A problem occurred evaluating project ':react-native-safe-area-context'.
> Project with path ':ReactAndroid' could not be found in project ':react-native-safe-area-context'.
```

Repro/steps: https://github.com/leotm/RN071RC5/commits/master

nb: it's with few other libs too
- https://github.com/leotm/react-native-template-new-architecture/issues/1315

cc @tomekzaw + @software-mansion team looking at adding RN 0.71.0-rc.5 support too if helps
- https://github.com/software-mansion/react-native-reanimated/pull/3745
- https://github.com/software-mansion/react-native-gesture-handler/pull/2313 (in fabric example)
- https://github.com/software-mansion/react-native-screens/pull/1631 (in fabric example)

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -99,10 +99,6 @@ repositories {
 def kotlin_version = getExtOrDefault('kotlinVersion', project.properties['RNSAC_kotlinVersion'])
 
 dependencies {
-    if (isNewArchitectureEnabled()) {
-        implementation project(":ReactAndroid")
-    } else {
-        implementation 'com.facebook.react:react-native:+'
-    }
+    implementation 'com.facebook.react:react-native:+'
     implementation "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
 }

